library/ziti.c

```diff
@@ -1226,11 +1237,14 @@ static void api_session_cb(ziti_api_session *session, const ziti_error *err, voi
         if (errCode == ZITI_NOT_AUTHORIZED) {
             if (ztx->api_session || !init_req->start) {
                 ZTX_LOG(DEBUG, "received NOT_AUTHORIZED attempting re-auth");
+                ziti_set_unauthenticated(ztx);
                 ziti_re_auth(ztx);
                 errCode = ztx->ctrl_status; // do not trigger event yet
             } else {
                 // cannot login or re-auth -- identity no longer valid
                 // notify service removal, and state
+                ziti_set_impossible_to_authenticate(ztx);
```

Comment:
A new API session state transition that means we are giving up on this identity.